pointers.go

```diff
@@ -0,0 +1,12 @@
+package geo
+
+// A Pointer is the interface for something that has a point.
+type Pointer interface {
+	// Point should return the "center" or other canonical point
+	// for the object. The caller is expected to Clone
+	// the point if changes need to be make.
+	Point() *Point
```

Comment:
Should this not be a pointer if we don't want to allow callers to manipulate the returned object? Maybe this question will be answered as I continue reviewing.